src/intervals/interval_rounding.jl

```diff
@@ -0,0 +1,193 @@
+
+## Macros for directed rounding:
+
+# Use the following empty definitions for rounding types other than Float64:
+Base.set_rounding(whatever, rounding_mode) = ()
+Base.get_rounding(whatever) = ()
+
+if VERSION > v"0.4-"
+    Base.Rounding.get_rounding_raw(whatever) = ()
+    Base.Rounding.set_rounding_raw(whatever, rounding_mode) = ()
+end
+
+
+const INTERVAL_ROUNDING = [:narrow]  # or :wide
+# TODO: replace by an enum in 0.4
+
+@doc doc"""`get_interval_rounding()` returns the current interval rounding mode.
+There are two possible rounding modes:
+
+- :narrow  -- changes the floating-point rounding mode to `RoundUp` and `RoundDown`.
+This gives the narrowest possible interval.
+
+- :wide -- Leaves the floating-point rounding mode in `RoundNearest` and uses
+`prevfloat` and `nextfloat` to achieve directed rounding. This creates an interval of width 2`eps`.
+""" ->
+
+get_interval_rounding() = INTERVAL_ROUNDING[end]
+
+set_interval_rounding(mode) = INTERVAL_ROUNDING[end] = mode  # a symbol
```

Comment:
I think it is a good idea to add checks to `mode` so it only accepts (for the time being) `:narrow` or `:wide`?